src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractDatabaseEngine.java

```diff
@@ -1382,12 +1461,24 @@ public synchronized ResultIterator iterator(String query, int fetchSize, int rea
     protected abstract void addColumn(final DbEntity entity, final DbColumn... columns) throws DatabaseEngineException;
 
     /**
-     * Adds the FKs.
+     * Adds all foreign keys.
      *
      * @param entity The entity.
      * @throws DatabaseEngineException If something goes wrong creating the FKs.
      */
-    protected abstract void addFks(final DbEntity entity) throws DatabaseEngineException;
+    protected void addFks(final DbEntity entity) throws DatabaseEngineException {
+        addFks(entity, new HashSet<>(entity.getFks()));
+    }
+
+    /**
+     * Adds foreign keys the foreign keys passed as argument to the entity.
+     *
+     * @param entity The entity             .
+     * @param fks    The foreign keys to be added.
+     *
+     * @throws DatabaseEngineException If something goes wrong creating the FKs.
+     */
+    protected abstract void addFks(final DbEntity entity, final Set<DbFk> fks) throws DatabaseEngineException;
```

Comment:
Is this a breaking change?
If `AbstractDatabaseEngine` is public api, then adding this method will break implementations of it.
Should we add a default implementation here (even if it does nothing, or throws a `NotImplementedException`) ?

Reply:
I think we're considering only the `DatabaseEngine` as part of the public api, not the abstract class or the other engine implementation classes.